utils/land-patterns.stanza

```diff
@@ -5,281 +5,406 @@ defpackage ocdb/land-patterns :
   import math
   import jitx
   import jitx/fonts
-  import jitx/commands
+  import jitx/commands with:
+    prefix(pad) => def-
+  import ocdb/design-vars
 
+;==============================================================================
+;============================== Helpers =======================================
+;==============================================================================
 ; ====== Argument constraint Helpers ===========================================
+; Ensure that a value is even and positive.
 defn ensure-even-positive! (value: Int, name: String):
   if value < 2 or value % 2 == 1:
     fatal("%_ only supports a positive even number of pins." % [name])
+  
+; Ensure that a value is a proper pad shape
+defn ensure-pad-shape! (s:Shape, name:String) -> Shape:
+  if s is-not Shape:
+    fatal("%_ is not a valid shape for pad geometry: %_." % [name, s])
+  s as Shape
 
 ; ====== Reference Label Helpers ===============================================
-public defn ref-label (x:Double, y:Double, a:Anchor) :
-  inside pcb-landpattern :
-    layer(Silkscreen("values", Top)) = Text(">REF", 0.7056, a, loc(x,y))
-    
+; Returns a reference label text
+public defn ref-label-text (pose:Pose, anchor:Anchor) -> Text:
```

Comment:
To avoid introducing too many concepts to the user, please un-refactor (defactor?) this back into the `ref-label` helper. Here is the reasoning:

If the only function we provide is `ref-label`, then we can provide a clear explanation of what it does, and there isn't anything for the user to complain about.

If we provide two functions: `ref-label` and `ref-label-text`, the user will ask the very reasonable questions:
1) What is the difference between the two?
2) When should I use one versus the other?
3) Do you recommend I follow this style (and factoring) for all of my own utilities as well?

And that is a lot of extra complexity without much gain.